### PR TITLE
Bump axe-core to v4.5.1

### LINKED ISSUE #250: Bump axe-core to version to v4.5.1
Hello!

Axe core v4.5.1 brings some bugfixes and some major performance improvements such as https://github.com/dequelabs/axe-core/pull/3423 that our project would benefit from. 

I'm happy to attach a PR.

### PR DESCRIPTION
**Pull request by jbchr**
Closes #250

### CODE REVIEW

**Review by AriPerkkio — CHANGES_REQUESTED** [review]
I would suggest to wait for next patch release due to https://github.com/dequelabs/axe-core/issues/3753.

You can see those errors by adding some `<style>` tags here:

https://github.com/NickColley/jest-axe/blob/208e197989398a2542f566871d900aacdc771cdb/__tests__/index.test.js#L24-L32

```diff
    const goodHtmlExample = `
     <html>
+       <head>
+         <style>
+          a { color: red; }
+         </style>
+       </head>
       <body>
```

Warnings seen in your branch with those changes:

```
PASS __tests__/index.test.js
  ● Console

    console.warn
      Couldn't load preload assets:  TypeError: Cannot read properties of undefined (reading 'mediaText')
          at /workspaces/jest-axe/node_modules/axe-core/axe.js:12577:47
          at Array.filter (<anonymous>)
          at getStylesheetsFromDocument (/workspaces/jest-axe/node_modules/axe-core/axe.js:12576:47)
          at getStylesheetsOfRootNode (/workspaces/jest-axe/node_modules/axe-core/axe.js:12557:18)
          at /workspaces/jest-axe/node_modules/axe-core/axe.js:12527:22
          at Array.forEach (<anonymous>)
          at getCssomForAllRootNodes (/workspaces/jest-axe/node_modules/axe-core/axe.js:12525:17)
          at Object.preloadCssom [as cssom] (/workspaces/jest-axe/node_modules/axe-core/axe.js:12502:14)
          at /workspaces/jest-axe/node_modules/axe-core/axe.js:12692:44
          at Array.map (<anonymous>)
          at /workspaces/jest-axe/node_modules/axe-core/axe.js:12691:28
          at new Promise (<anonymous>)
          at preload (/workspaces/jest-axe/node_modules/axe-core/axe.js:12686:14)
          at /workspaces/jest-axe/node_modules/axe-core/axe.js:22426:15
          at pop (/workspaces/jest-axe/node_modules/axe-core/axe.js:6501:18)
          at Object.defer (/workspaces/jest-axe/node_modules/axe-core/axe.js:6523:11)
          at Audit.run (/workspaces/jest-axe/node_modules/axe-core/axe.js:22425:28)
          at /workspaces/jest-axe/node_modules/axe-core/axe.js:22710:15
          at pop (/workspaces/jest-axe/node_modules/axe-core/axe.js:6501:18)
          at Object.defer (/workspaces/jest-axe/node_modules/axe-core/axe.js:6523:11)
          at Object.runRules [as _runRules] (/workspaces/jest-axe/node_modules/axe-core/axe.js:22709:9)
          at Object.run (/workspaces/jest-axe/node_modules/axe-core/axe.js:22958:11)
          at /workspaces/jest-axe/index.js:68:15
          at new Promise (<anonymous>)
          at axe (/workspaces/jest-axe/index.js:67:12)
          at Object.<anonymous> (/workspaces/jest-axe/__tests__/index.test.js:68:13)
          at Object.asyncJestTest (/workspaces/jest-axe/node_modules/jest-jasmine2/build/jasmineAsyncInstall.js:106:37)
          at /workspaces/jest-axe/node_modules/jest-jasmine2/build/queueRunner.js:45:12
          at new Promise (<anonymous>)
          at mapper (/workspaces/jest-axe/node_modules/jest-jasmine2/build/queueRunner.js:28:19)
          at /workspaces/jest-axe/node_modules/jest-jasmine2/build/queueRunner.js:75:41

      at node_modules/axe-core/axe.js:22429:25
```

File: package.json [hunk 1]
```diff
@@ -24,7 +24,7 @@
     "node": ">= 10.0.0"
   },
   "dependencies": {
-    "axe-core": "4.4.1",
+    "axe-core": "4.5.0",
```
Comment (AriPerkkio) [inline package.json:27]: This is now released and includes fix for https://github.com/dequelabs/axe-core/issues/3753.

```suggestion
    "axe-core": "4.5.1",
```

**Review by AriPerkkio — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (jbchr) [reply package.json]: @AriPerkkio thank you for the update :) I updated the PR to also include the corresponding change in the `package-lock.json`

**Review by AriPerkkio — APPROVED** [review]

**Comment by NickColley** [comment]
Heya folks, thanks for waiting til that patch came through, I'll get an update out soon!

**Comment by NickColley** [comment]
Will sort this on your behalf, dont worry about updating this pr

**Pull request closed without merging**